### ISSUE TITLE
Unreachable code not removed

### ISSUE THREAD

**Issue by Andarist** [rollup/rollup#2319]
[Reproduction](https://rollupjs.org/repl?version=0.62.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiolMjBUUkVFLVNIQUtJTkclMjAqJTJGJTVDbmltcG9ydCUyMCU3QiUyMGNhbGN1bGF0ZUhlaWdodCUyMCU3RCUyMGZyb20lMjAnLiUyRmRvbS5qcyclM0IlNUNuJTVDbmV4cG9ydCUyMGRlZmF1bHQlMjBjbGFzcyUyMEElMjAlN0IlNUNuJTIwJTIwcmVzaXplKCklMjAlN0IlNUNuJTVDdCU1Q3QlMkYlMkYlMjB0cnVlJTIwaW5saW5lZCUyMGJ5JTIwcm9sbHVwLXBsdWdpbi1yZXBsYWNlJTVDbiU1Q3QlMjAlMjBpZiUyMCh0cnVlKSUyMHJldHVybiU1Q24lNUN0JTVDdCU1Q24lNUN0JTVDdHRoaXMuc2V0U3RhdGUoJTdCJTIwaGVpZ2h0JTNBJTIwY2FsY3VsYXRlSGVpZ2h0KHRoaXMuX3JlZiklMjAlN0QpJTVDbiU1Q3QlN0QlNUNuJTdEJTVDbiUyMiU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJkb20uanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyJTVDbmV4cG9ydCUyMGZ1bmN0aW9uJTIwY2FsY3VsYXRlSGVpZ2h0JTIwKCUyMG5vZGUlMjApJTIwJTdCJTVDbiU1Q3RyZXR1cm4lMjBub2RlLmNsaWVudEhlaWdodCU1Q24lN0QlNUNuJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE)

Basically unused functions (and probably other things too) do not get tree-shaken if they are unreachable

**Comment by lukastaegert** [comment]
At the moment, rollup does not handle early returns but there are several ways of doing this. I hope we can improve this at some point.

**Comment by Andarist** [comment]
Oh, I havent realized early returns are a special case. If there is some other issue tracking this, feel free to close this one.

**Comment by lukastaegert** [comment]
As a matter of fact I think this is the first issue about this :)

**Comment by shellscape** [comment]
Hey folks. This is a saved-form message, but rest assured we mean every word. The Rollup team is attempting to clean up the Issues backlog in the hopes that the active and still-needed, still-relevant issues bubble up to the surface. With that, we're closing issues that have been open for an eon or two, and have gone stale like pirate hard-tack without activity.

We really appreciate the folks have taken the time to open and comment on this issue. Please don't confuse this closure with us not caring or dismissing your issue, feature request, discussion, or report. The issue will still be here, just in a closed state. If the issue pertains to a bug, please re-test for the bug on the latest version of Rollup and if present, please tag @shellscape and request a re-open, and we'll be happy to oblige.

**Comment by lukastaegert** [comment]
I hope to be able to address this one soon.
